src/clj_kondo/impl/namespace.clj

```diff
@@ -464,6 +464,45 @@
           sym))
       sym)))
 
+(defn generated?
+  "We assume that a node is generated by clj-kondo when it has no metadata or
+  when it has the key :clj-kondo.impl/generated. For this purpose, we're only
+  checking the whole expression or the first child (in the case of a list node)."
+  [expr]
+  (or (nil? (meta expr))
+      (:clj-kondo.impl/generated expr)
+      (some-> expr :children first :clj-kondo.impl/generated)))
+
+(defn lint-existing-alias
+  "Check if the given symbol is qualified and isn't using an `:as` or
+  `:as-alias` alias."
+  [ctx expr ns-sym]
+  ;; Reasons to not check the given symbol:
+  ;; * it's already using an existing alias
+  ;; * this lint is disabled
+  ;; * this lint is configured to exclude the given namespace
+  ;; * it's generated by clj-kondo
+  (when-not (or (contains? (:aliases (:ns ctx)) ns-sym)
+                (linter-disabled? ctx :existing-alias)
+                (config/existing-alias-excluded? (:config ctx) ns-sym)
+                (generated? expr))
+    (when-let [ns->aliases (:ns->aliases (:ns ctx))]
```

Comment:
@borkdude from here

Reply:
Ah right, I only read the commit message, not the actual commit ;).